jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -927,6 +927,24 @@ default int size()
         return size;
     }
 
+    /**
+     * @param fields the {@link HttpFields} to convert to a {@link Map}.
+     * @return an unmodifiable {@link Map} representing the contents of the {@link HttpFields}.
+     */
+    static Map<String, List<String>> asMap(HttpFields fields)
+    {
+        return new HttpFieldsMap.Immutable(fields);
+    }
+
+    /**
+     * @param fields the {@link HttpFields} to convert to a {@link Map}.
+     * @return a {@link Map} where changes to the contents will be reflected in the supplied {@link HttpFields}.
+     */
+    static Map<String, List<String>> asMutableMap(HttpFields.Mutable fields)
+    {
+        return new HttpFieldsMap.Mutable(fields);
+    }
+
```

Comment:
Meh!

```suggestion
    /**
     * @param fields the {@link HttpFields} to convert to a {@link Map}.
     * @return an unmodifiable {@link Map} representing the contents of the {@link HttpFields}.
     */
    static Map<String, List<String>> asMap(HttpFields fields)
    {
        return (fields instanceof HttpFields.Mutable mutable)
            ? new HttpFieldsMap.Mutable(mutable);
            : new HttpFieldsMap.Immutable(fields);
    }
```

This give mutable for mutable, immutable for immutable and you don't have to worry about the actual type.   We already do similar instanceof checks in MutableHttpFields, so best to follow that pattern

Reply:
I'll do this, but I do think it makes the code a lot more confusing.

Before you could easily view where each method is called to see where we allow the application to modify the headers, now it is not possible.

Reply:
Done.